racun-service/src/main/java/rs/edu/raf/banka/racun/controller/UgovorController.java

```diff
@@ -0,0 +1,163 @@
+        try {
+            List<Ugovor> ugovori = ugovorService.getAllByCompany(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+
+        } catch (Exception ex) {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @GetMapping(value = "/get/company/{kompanijaId}/finalized", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllKompanijaFinalized(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
+        try
+        {
+            var ugovori = ugovorService.getAllByCompanyFinalized(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @GetMapping(value = "/get/company/{kompanijaId}/draft", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllKompanijaDraft(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
+        try
+        {
+            var ugovori = ugovorService.getAllByCompanyDraft(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PostMapping(value = "/create", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> createUgovor(@RequestHeader("Authorization") String token, @RequestBody UgovorCreateRequest request) {
+        try
+        {
+            var ugovor = ugovorService.createUgovor(request);
+            return ResponseEntity.ok(ugovor);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PutMapping(value = "/modify", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> modifyUgovor(@RequestHeader("Authorization") String token, @RequestBody UgovorUpdateRequest request) {
+        try
+        {
+            var result = ugovorService.modifyUgovor(request);
+            return ResponseEntity.ok(result);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PutMapping(value = "/modify/{id}/addDocument", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> modifyUgovorDocument(@RequestHeader("Authorization") String token, @PathVariable Long id, @RequestParam("file") MultipartFile file) {
+        try {
+            var result = ugovorService.modifyDocument(id, file);
+            return ResponseEntity.ok(result);
+        } catch (Exception ex) {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PutMapping(value = "/modify/addStavka", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Promeni rutu u nešto poput `/stavka` i neka bude POST zahtev.

Reply:
Mozda bih ovde stavio /transakcija ili /transakcionaStavka, ovako ce mozda biti malo nejasno frontu.